### PR TITLE
feat: [PL-23648]: OAS for non-compalaint components in Platform service

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33026)

### CODE REVIEW

**Comment by VikasMaddukuriHarness** [comment]
trigger build_time_check-code_format

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/AccountNotificationSettingDTO.java [hunk 1]
```diff
@@ -9,14 +9,17 @@
...
 import lombok.Builder;
 import lombok.Data;
 
 @Data
 @Builder
+@Schema(name = "AccountNotificationSetting", description = "This has the Account Notification settings.")
 public class AccountNotificationSettingDTO {
-  @NotNull String accountId;
+  @Schema(description = "Account Identifier.") @NotNull String accountId;
+  @Schema(description = "Specify whether or not to send notifications via Delegate.")
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/AccountNotificationSettingDTO.java:22]: Specify if notifications should be sent through Delegate.

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/EmailSettingDTO.java [hunk 2]
```diff
@@ -18,9 +19,10 @@
 @Data
 @NoArgsConstructor
 @EqualsAndHashCode(callSuper = true)
+@Schema(name = "EmailSettings", description = "This has the EmailSetting details defined in Harness.")
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/EmailSettingDTO.java:22]: Details of the Email settings configured in Harness.

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/MSTeamSettingDTO.java [hunk 3]
```diff
@@ -17,6 +18,7 @@
 @Data
 @NoArgsConstructor
 @EqualsAndHashCode(callSuper = true)
+@Schema(name = "MSTeamSetting", description = "This has the MSTeam setting details defined in Harness")
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/MSTeamSettingDTO.java:21]: Details of the MS Teams settings configured in Harness.

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationDTO.java [hunk 4]
```diff
@@ -19,11 +20,14 @@
 @Data
 @Builder
 @FieldDefaults(level = AccessLevel.PRIVATE)
+@Schema(name = "Notification", description = "This has the Notification details defined in Harness.")
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationDTO.java:23]: Does this look right?
Details of the Notification settings configured in Harness.

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationDTO.java [hunk 5]
```diff
@@ -19,11 +20,14 @@
...
-  Team team;
+  @Schema(description = "Identifier of the notification.") String id;
+  @Schema(description = "Account Identifier.") String accountIdentifier;
+  @Schema(description = "Team associated with the notification.") Team team;
+  @Schema(description = "Channel type of notification. We currently support SLACK, EMAIL, PAGERDUTY and MSTEAMS.")
   NotificationChannelType channelType;
+  @Schema(description = "Boolean responses of whether or not the notification is sent.")
   List<Boolean> processingResponses;
-  int retries;
+  @Schema(description = "Number of retries done to send the notification.") int retries;
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationDTO.java:32]: The number of times the notification was resent.

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationSettingDTO.java [hunk 6]
```diff
@@ -34,10 +35,11 @@
       @JsonSubTypes.Type(value = PagerDutySettingDTO.class, name = "PAGERDUTY"),
 })
 @JsonIgnoreProperties(ignoreUnknown = true)
+@Schema(name = "NotificationSetting", description = "This has the Notification setting details defined in Harness")
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationSettingDTO.java:38]: Please change as per earlier comment.

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/PagerDutySettingDTO.java [hunk 7]
```diff
@@ -17,6 +18,7 @@
 @Data
 @NoArgsConstructor
 @EqualsAndHashCode(callSuper = true)
+@Schema(name = "PagerDutySetting", description = "This has the PagerDuty settings defined in Harness")
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/PagerDutySettingDTO.java:21]: Details of the Pager Duty settings configured in Harness.

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/SlackSettingDTO.java [hunk 8]
```diff
@@ -17,6 +18,7 @@
 @Data
 @NoArgsConstructor
 @EqualsAndHashCode(callSuper = true)
+@Schema(name = "SlackSetting", description = "This has the Slack settings defined in Harness")
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/SlackSettingDTO.java:21]: Details of the Slack settings configured in Harness.

File: 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/TemplateDTO.java [hunk 9]
```diff
@@ -9,15 +9,17 @@
 
 import io.harness.notification.Team;
 
+import io.swagger.v3.oas.annotations.media.Schema;
 import lombok.Builder;
 import lombok.Data;
 
 @Data
 @Builder
+@Schema(name = "Template", description = "This has the Template details defined in Harness")
```
Comment (rashminandasahoo) [inline 950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/TemplateDTO.java:18]: Details of the Templates configured in Harness.

File: platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/beans/AuditSettingsDTO.java [hunk 10]
```diff
@@ -6,18 +6,19 @@
...
 import lombok.Builder;
 import lombok.Data;
 
 @OwnedBy(PL)
 @Data
 @Builder
+@Schema(name = "AuditSettings", description = "This has the AuditSettings details defined in Harness.")
 public class AuditSettingsDTO {
-  @NotNull int retentionPeriodInMonths;
+  @Schema(description = "Retention time of the Audits in terms of months") @NotNull int retentionPeriodInMonths;
```
Comment (rashminandasahoo) [inline platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/beans/AuditSettingsDTO.java:23]: Retention of Audit Logs in months.

File: platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/beans/YamlDiffRecordDTO.java [hunk 11]
```diff
@@ -11,13 +11,15 @@
...
 import lombok.Data;
 
 @OwnedBy(PL)
 @Data
 @Builder
+@Schema(name = "YamlDiffRecord", description = "This has the Yaml Difference of the record.")
 public class YamlDiffRecordDTO {
-  String oldYaml;
-  String newYaml;
+  @Schema(description = "previous Yaml.") String oldYaml;
```
Comment (rashminandasahoo) [inline platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/beans/YamlDiffRecordDTO.java:23]: Can we say "Previous YAML version" and "Current YAML version"?
Also please correct the casing of YAML.

**Review by VikasMaddukuriHarness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33026)* status: 15 file(s) to review. 11. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/AccountNotificationSettingDTO.java` line 22 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4u_4hUJbWY7VX6-r2-22:-N2z5lBP0g-MWA651uhU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/AccountNotificationSettingDTO.java#L22)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Specify if notifications should be sent through Delegate.
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/EmailSettingDTO.java` line 22 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4vH7Bsu6J6os9o-r2-22:-N2z5vYr7TleVb7JHmEw:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/EmailSettingDTO.java#L22)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Details of the Email settings configured in Harness.
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/MSTeamSettingDTO.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4vH7Bsu6J6os9p-r2-21:-N2z65JhDmXuo5briCA4:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/MSTeamSettingDTO.java#L21)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Details of the MS Teams settings configured in Harness.
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationDTO.java` line 23 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4vH7Bsu6J6os9q-r2-23:-N2z6DW4EtbVbbBT65UU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationDTO.java#L23)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Does this look right?
Details of the Notification settings configured in Harness.
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationDTO.java` line 32 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4vH7Bsu6J6os9q-r2-32:-N2z6Ie27JL33f3sZDVW:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationDTO.java#L32)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

The number of times the notification was resent.
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationSettingDTO.java` line 38 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4vH7Bsu6J6os9r-r2-38:-N2z6rMX4xMa4V6xQLdN:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/NotificationSettingDTO.java#L38)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Please change as per earlier comment.
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/PagerDutySettingDTO.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4vH7Bsu6J6os9s-r2-21:-N2z70UaAZUr-lKSbWYJ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/PagerDutySettingDTO.java#L21)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Details of the Pager Duty settings configured in Harness.
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/SlackSettingDTO.java` line 21 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4vH7Bsu6J6os9t-r2-21:-N2z74phEVyOTJ0I05UG:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/SlackSettingDTO.java#L21)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Details of the Slack settings configured in Harness.
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/TemplateDTO.java` line 18 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N2QmO4vH7Bsu6J6os9u-r2-18:-N2z792s6GFHK9oPcrGT:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/950-delegate-tasks-beans/src/main/java/io/harness/notification/remote/dto/TemplateDTO.java#L18)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Details of the Templates configured in Harness.
</blockquote></details>

Done.
___
*[`platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/beans/AuditSettingsDTO.java` line 23 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N1lmlImd6738p6TtMC9-r2-23:-N2z7Gnz70HFy7rIp5T5:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/beans/AuditSettingsDTO.java#L23)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Retention of Audit Logs in months.
</blockquote></details>

Done.
___
*[`platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/beans/YamlDiffRecordDTO.java` line 23 at r2](https://reviewable.io/reviews/harness/harness-core/33026#-N1lmlImd6738p6TtMCB-r2-23:-N2z7abtAd06PqlR7gFR:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/e7264be5c19339c5fcf7bc03af2e243bc44f07ed/platform-service/modules/audit-service/contracts/src/main/java/io/harness/audit/beans/YamlDiffRecordDTO.java#L23)):*
<details><summary><i>Previously, rashminandasahoo wrote…</i></summary><blockquote>

Can we say "Previous YAML version" and "Current YAML version"?
Also please correct the casing of YAML.
</blockquote></details>

Done.

**Review by rashminandasahoo — APPROVED** [review]
:lgtm:

> Reviewed 1 of 7 files at r1, 4 of 13 files at r2, 10 of 10 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33026#-:-N2z8hJZ-IvFXidkr7SF:bnfp4nl)* status: 15 file(s) to review. 1 of 2 approvals obtained.

**Review by meenakshi-harness — APPROVED** [review]
> Reviewed 1 of 7 files at r1, 4 of 13 files at r2, 10 of 10 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33026)* status: :shipit: complete! ✓

**Pull request merged**